core/src/main/java/com/datastax/oss/driver/internal/core/metadata/SniEndPoint.java

```diff
@@ -64,7 +64,7 @@ public InetSocketAddress resolve() {
       // The order of the returned address is unspecified. Sort by IP to make sure we get a true
       // round-robin
       Arrays.sort(aRecords, IP_COMPARATOR);
-      int index = (aRecords.length == 1) ? 0 : (int) OFFSET.getAndIncrement() % aRecords.length;
+      int index = (aRecords.length == 1) ? 0 : Math.abs((int) OFFSET.getAndIncrement()) % aRecords.length;
```

Comment:
Why is `OFFSET` an `AtomicLong`, shouldn't it be an `AtomicInteger`?

Reply:
Good catch, changing.